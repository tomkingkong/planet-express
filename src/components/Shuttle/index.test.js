import React from 'react';
import { shallow } from 'enzyme';

import { Shuttle } from '.';

describe('Shuttle component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Shuttle />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should allow the user to proceed to the ISS', () => {
    const mockFn = jest.fn();
    const history = { replace: mockFn };
    const wrapper = shallow(<Shuttle history={history}/>);
    wrapper.find('.continue_btn').simulate('click');
    expect(mockFn).toHaveBeenCalled();
  });
});