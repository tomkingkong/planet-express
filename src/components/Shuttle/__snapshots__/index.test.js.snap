// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shuttle component should match snapshot 1`] = `
<div
  className="Shuttle"
>
  <div
    className="shuttle__launch_img"
    style={
      Object {
        "backgroundImage": "url(/images/shuttle/shuttle_takeoff_smokey.jpg)",
      }
    }
  />
  <div
    className="shuttle__info"
  >
    <h4
      className="shuttle__headline"
    >
      Departing Earth..
    </h4>
    <p
      className="shuttle__desc"
    >
      Launching from Cape Canaveral on 
        Sun Sep 23 2018, 
        headed for rendezvous with the Internation Space Station.
    </p>
  </div>
  <button
    className="continue_btn"
    onClick={[Function]}
  >
    continue
  </button>
</div>
`;
